### PR TITLE
Code coverage reports using codecov.io

### PR DESCRIPTION
**Pull request by psturc**
# Issue link
https://issues.redhat.com/browse/MGDAPI-2683

# What
* add code coverage [badge](https://github.com/psturc/cloud-resource-operator/tree/MGDAPI-2683#readme)
* prepare for coverage reports on PRs via prow
* removed travis.ci config

# Verification steps
When run locally, following command should pass and generate `coverage.out` report
```bash
make coverage
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -132,28 +132,12 @@ cluster/clean:
...
 	GO111MODULE=off go get -u github.com/rakyll/gotest
 	gotest -v -covermode=count -coverprofile=coverage.out ./pkg/providers/... ./pkg/resources/... ./apis/integreatly/v1alpha1/types/... ./pkg/client/...
 
-.PHONY: test/unit/coverage
-test/unit/coverage:
-	@echo Running the coverage cli and html
-	go tool cover -html=coverage.out
-	go tool cover -func=coverage.out
-
-.PHONY: test/unit/ci
```
Comment (psturc) [inline Makefile:152]: used previously by travis

File: Makefile [hunk 2]
```diff
@@ -132,28 +132,12 @@ cluster/clean:
 	@$(KUSTOMIZE) build config/rbac | oc delete --force -f -	
 	oc delete project $(NAMESPACE)
 
-.PHONY: test/unit/setup
```
Comment (psturc) [inline Makefile:135]: used previously by travis

**Comment by KevFan** [comment]
Verified `coverage.out` file was generated :+1: 

image

/lgtm

**Pull request merged**